src/admin/views/Menu/MenuDetail.tsx

```diff
@@ -0,0 +1,244 @@
+import React, { FunctionComponent, useEffect, useRef, useState } from 'react';
+import { connect } from 'react-redux';
+import { RouteComponentProps, withRouter } from 'react-router';
+
+import { useMutation } from '@apollo/react-hooks';
+import { Button, ButtonToolbar, Flex, IconName, Spacer, Table } from '@viaa/avo2-components';
+import { cloneDeep, isEqual, startCase } from 'lodash-es';
+import { Dispatch } from 'redux';
+
+import { DataQueryComponent } from '../../../shared/components/DataComponent/DataQueryComponent';
+import { buildLink } from '../../../shared/helpers/generateLink';
+import { ApolloCacheManager } from '../../../shared/services/data-service';
+import toastService, { TOAST_TYPE } from '../../../shared/services/toast-service';
+import { GET_MENU_ITEMS_BY_PLACEMENT, UPDATE_MENU_ITEM_BY_ID } from '../../graphql';
+import { AdminLayout, AdminLayoutActions, AdminLayoutBody } from '../../layouts';
+import { ADMIN_PATH } from '../../routes';
+import { setMenuItems } from '../../store/actions';
+import { MenuItem } from '../../types';
+
+import './MenuDetail.scss';
+
+interface MenuDetailProps extends RouteComponentProps<{ menu: string }> {
+	setMenuItemsAction: (menuItems: MenuItem[]) => void;
+}
+
+const MenuDetail: FunctionComponent<MenuDetailProps> = ({ history, match, setMenuItemsAction }) => {
+	const [activeRow, setActiveRow] = useState<number | null>(null);
+	const [isSaving, setIsSaving] = useState<boolean>(false);
+	const [initialMenuItems, setInitialMenuItems] = useState<MenuItem[]>([]);
+	const [menuItems, setMenuItems] = useState<MenuItem[]>([]);
+
+	const [updateMenuItem] = useMutation(UPDATE_MENU_ITEM_BY_ID);
+
+	const hasInitialData = useRef<boolean>(false);
+	const timeout = useRef<NodeJS.Timeout | null>(null);
+
+	useEffect(() => {
+		if (menuItems && menuItems.length) {
+			// Save menu items in store
+			setMenuItemsAction(menuItems);
+		}
+	}, [menuItems, setMenuItemsAction]);
+
+	useEffect(() => {
+		// Reset active row to clear styling
+		timeout.current = setTimeout(() => {
+			setActiveRow(null);
+		}, 1000);
+
+		return () => {
+			if (timeout.current) {
+				clearTimeout(timeout.current);
+			}
+		};
+	}, [activeRow]);
+
+	// Computed
+	const menuId = match.params.menu;
+
+	// Methods
+	const handleNavigate = (path: string, params: { [key: string]: string } = {}): void => {
+		history.push(buildLink(path, params));
+	};
+
+	const handleSave = (refetch: () => void): void => {
+		setIsSaving(true);
+
+		const promises: Promise<any>[] = [];
+		menuItems.forEach(menuItem => {
+			promises.push(
+				updateMenuItem({
```

Comment:
Overal voor graphql mutations gebruiken we het formaat: <trigger><subject><operation>, dus dit mss hernoemen naar triggerMenuItemUpdate